python/rmm/rmm/librmm/memory_resource.pxd

```diff
@@ -120,17 +120,16 @@ cdef extern from "rmm/mr/device/cuda_async_view_memory_resource.hpp" \
             cudaMemPool_t pool_handle) except +
         cudaMemPool_t pool_handle() const
 
-# TODO: when we adopt Cython 3.0 use enum class
 cdef extern from "rmm/mr/device/cuda_async_memory_resource.hpp" \
         namespace \
-        "rmm::mr::cuda_async_memory_resource::allocation_handle_type" \
+        "rmm::mr::cuda_async_memory_resource" \
         nogil:
-    enum allocation_handle_type \
-            "rmm::mr::cuda_async_memory_resource::allocation_handle_type":
+    cpdef enum class allocation_handle_type(int32_t):
```

Comment:
We should probably explicitly declare https://github.com/rapidsai/rmm/blob/1226490cf44741e949db65fe5910c13a1e4fb5eb/cpp/include/rmm/mr/device/cuda_async_memory_resource.hpp#L58 as an `int32_t` so we are guaranteed a match here. An `enum class` defaults to `int` but I wouldn't want the C++ definition to change unexpectedly. An explicit default makes that clearer.

Reply:
I'm not very familiar with Sphinx, it was [raising a warning that's treated as error](https://github.com/rapidsai/rmm/actions/runs/15072047396/job/42382291092?pr=1787#step:10:1725). I couldn't find a different way to fix it other than ignoring it in https://github.com/rapidsai/rmm/pull/1787/commits/569ea8838e5168c79abca432800e41063869b33c. If you know of a better way to resolve that please let me know and I'll make the change.